src/mlpack/methods/preprocess/preprocess_split_main.cpp

```diff
@@ -148,6 +148,9 @@ void BINDING_FUNCTION(util::Params& params, util::Timers& timers)
       [](double x) { return x >= 0.0 && x <= 1.0; }, true,
       "test ratio must be between 0.0 and 1.0");
 
+  // input data can't be empty.
+  RequireNonEmptyInputValue(params, "input", true, "can not be empty");
```

Comment:
The user will simply get the message `input can not be empty`, which is true, but not particularly helpful.  It's better if we can tell the user in an error *why* their code failed.  So I would suggest a more ideal error message that the user might encounter would be something along the lines of `matrix "input" must be non-empty: cannot split an empty dataset` or similar.